src/components/UpcomingQuizCard/UpcomingQuizCard.utils.ts

```diff
@@ -0,0 +1,10 @@
+import { format } from 'date-fns';
+
+import { DEFAULT_DATE_FORMAT } from './UpcomingQuizCard.constants';
+
+export const formatDate = (
+  date: string | Date,
+  formatStr: string = DEFAULT_DATE_FORMAT
+): string => {
+  return format(new Date(date), formatStr);
```

Comment:
formatDate uses `new Date(date)` which can produce inconsistent results for ISO strings and does not validate the input. Prefer parseISO for YYYY-MM-DD and provide locale options when formatting.

**Sugerencia:**
```typescript
Use date-fns parseISO for ISO date strings and provide optional locale support. Example:

import { parseISO } from 'date-fns';
const d = typeof date === 'string' ? parseISO(date) : date;
return format(d, formatStr);

Also guard against invalid dates (isValid from date-fns) and return an empty string or fallback.
```